### PR TITLE
Editing structure of code styles

### PR DESCRIPTION
**Pull request by NikitaBondarevv**

### CODE REVIEW

File: src/components/header/authorizedUser/index.tsx [hunk 1]
```diff
@@ -29,6 +30,7 @@ export const AuthorizedUser = () => {
     <div className={styles.menu}>
       <Link to="/" className={styles.userName}>
         {user?.firstName}
+        <UserIcon/>
```
Comment (iliatcymbal) [inline src/components/header/authorizedUser/index.tsx]: space before closing bracket

File: src/components/header/svgIcons/menu.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      <g id="SVGRepo_iconCarrier">
+         <g id="Group_117" data-name="Group 117">
+            <rect id="Rectangle_56" data-name="Rectangle 56" width="256" height="64" transform="translate(128 320)" />
+            <rect id="Rectangle_57" data-name="Rectangle 57" width="256" height="64" transform="translate(128 224)" />
+            <path id="Path_58" data-name="Path 58" d="M480,0H32A31.981,31.981,0,0,0,0,32V480a31.981,31.981,0,0,0,32,32H480a31.981,31.981,0,0,0,32-32V32A31.981,31.981,0,0,0,480,0ZM448,448H64V64H448Z" />
+            <rect id="Rectangle_58" data-name="Rectangle 58" width="256" height="64" transform="translate(128 128)" />
+         </g>
+      </g>
+   </svg>
+)
```
Comment (iliatcymbal) [inline src/components/header/svgIcons/menu.tsx]: add an empty line in the end of file

File: src/components/publishItems/styles.css [hunk 3]
```diff
@@ -49,7 +34,7 @@
   row-gap: 40px;
 }
 
-.content ul {
+ul {
```
Comment (iliatcymbal) [inline src/components/publishItems/styles.css]: do not use global styles

File: src/components/publishItems/styles.css [hunk 4]
```diff
@@ -88,17 +73,17 @@
   cursor: pointer;
 }
 
-.content a {
+a {
   text-decoration: none;
   color: rgb(202, 202, 202);
 }
```
Comment (iliatcymbal) [inline src/components/publishItems/styles.css]: do not use global styles, or if it should be set on all links you have to put it into main styles

File: src/pages/createCategory/index.tsx [hunk 5]
```diff
@@ -4,14 +4,17 @@ import { getProducts } from 'contracts/products'
...
+import { EditableText } from 'components/editableText'
+import styles from './styles.css'
 
 export const CreateCategory = () => {
   const [allProducts, setAllProducts] = useState<IProduct[]>([])
   const products: IProduct[] = []
   const productsIds = useMemo(() => (products || []).map(product => product.id), [products])
+  const [valueEdit, setValueEdit] = useState('')
 
   console.log(productsIds);
```
Comment (iliatcymbal) [inline src/pages/createCategory/index.tsx]: remove

File: src/pages/styles.css [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  width: 550px;
+  min-height: 200px;
+  border-radius: 20px;
+  font-size: 25px;
+  padding: 80px 40px;
+  border: 1px solid rgba(113 119 144 / 25%);
+  color: rgb(202, 202, 202);
+  margin: 0 auto;
+  position: relative;
+}
```
Comment (iliatcymbal) [inline src/pages/styles.css]: empty line

File: src/components/header/authorizedUser/userIcon.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
+export const UserIcon = () => (
+  <svg
+    fill="#fff"
+    version="1.1"
+    id="Layer_1"
+    xmlns="http://www.w3.org/2000/svg"
+    xmlnsXlink="http://www.w3.org/1999/xlink"
+    width="25px" height="25px"
+    viewBox="0 0 512 512"
+    enable-background="new 0 0 512 512"
```
Comment (iliatcymbal) [inline src/components/header/authorizedUser/userIcon.tsx]: `enable-background` use appropriate jsx attribute

File: src/components/header/svgIcons/contacts.tsx [hunk 8]
```diff
@@ -0,0 +1,23 @@
+export const ContactsIcon = () => (
+  <svg width="25px" height="25px" viewBox="0 0 48 48" xmlns="http://www.w3.org/2000/svg" fill="#fff" stroke="none">
+    <g id="SVGRepo_bgCarrier" stroke-width="0" />
+    <g id="SVGRepo_tracerCarrier" stroke-linecap="round" stroke-linejoin="round" />
```
Comment (iliatcymbal) [inline src/components/header/svgIcons/contacts.tsx]: use appropriate jsx attributes

**Review by iliatcymbal — CHANGES_REQUESTED** [review]

File: src/pages/createCategory/index.tsx (hunk 5 above)
Comment (NikitaBondarevv) [reply src/pages/createCategory/index.tsx]: Why remove this?

File: src/components/publishItems/index.tsx [hunk 9]
```diff
@@ -22,7 +22,9 @@ export function PublishItems <T extends { id: number, title: string }>({
   create,
   onSave,
   filterPredicate,
-  valueEdit
+  valueEdit,
+  vievMode,
```
Comment (iliatcymbal) [inline src/components/publishItems/index.tsx]: view

File: src/components/publishItems/types.ts [hunk 10]
```diff
@@ -14,6 +14,8 @@ export type TPublishItemsProps<T> = {
   valueEdit?: string
   onSave?: (name: string) => Promise<void>
   filterPredicate: (data: T) => boolean
+  vievMode: boolean | undefined
```
Comment (iliatcymbal) [inline src/components/publishItems/types.ts]: `viewMode?: boolean `

File: src/components/publishItems/types.ts [hunk 11]
```diff
@@ -14,6 +14,8 @@ export type TPublishItemsProps<T> = {
...
+  getLink?: (data: T) => string
```
Comment (iliatcymbal) [inline src/components/publishItems/types.ts]: why is it optional?

File: src/pages/createCategory/index.tsx [hunk 12]
```diff
@@ -76,6 +76,7 @@ export const CreateCategory = () => {
         filterPredicate={getCategoryProducts}
         showEditButton
         valueEdit={valueEdit}
+        vievMode={true}
```
Comment (iliatcymbal) [inline src/pages/createCategory/index.tsx]: `={true}`no need

File: src/pages/products/index.tsx [hunk 13]
```diff
@@ -4,24 +4,27 @@ import { useParams } from 'react-router-dom'
...
+    setProduct(await findProducts(title))
+  }
 
+  useEffect(() => {
     getData()
   }, [])
 
-  console.log(product);
-
+  const onSave = async () => {
```
Comment (iliatcymbal) [inline src/pages/products/index.tsx]: handleSave

File: src/components/publishItems/types.ts (hunk 11 above)
Comment (NikitaBondarevv) [reply src/components/publishItems/types.ts]: Because this don't use in createCategory

File: src/pages/products/index.tsx [hunk 14]
```diff
@@ -4,24 +4,27 @@ import { useParams } from 'react-router-dom'
 import { IProduct } from 'interfaces/IProduct'
 import { UserContext } from 'contexts/userContext'
 import { EditableText } from 'components/editableText'
-import { findProducts } from 'contracts/products'
+import { findProducts, updateProduct } from 'contracts/products'
 import styles from './styles.css'
 
 export const Products = () => {
```
Comment (iliatcymbal) [inline src/pages/products/index.tsx:10]: why many products?

File: src/styles.css [hunk 15]
```diff
@@ -18,3 +18,20 @@
...
+:global a {
+  text-decoration: none;
+  color: rgb(202, 202, 202);
+}
+
+:global svg {
+  margin: 0 0 0 5px;
+  position: absolute;
+  right: -30px;
+}
```
Comment (iliatcymbal) [inline src/styles.css]: It seems not very good to set such styles for all svg

File: src/components/publishItems/types.ts (hunk 11 above)
Comment (iliatcymbal) [reply src/components/publishItems/types.ts]: then you shouldn't use `getLink!`

Comment (NikitaBondarevv) [reply src/components/publishItems/types.ts]: No, this is my mistake. In createCatigory this should be used too

**Pull request merged**